api/src/main/java/org/openmrs/module/pihcore/apploader/CustomAppLoaderFactory.java

```diff
@@ -2163,6 +2164,7 @@ private void enableMCHForms() {
                     and(sessionLocationHasTag("Maternal and Child Location"),
                             visitDoesNotHaveEncounterOfType(PihEmrConfigConstants.ENCOUNTERTYPE_ANC_INTAKE_UUID),
                             visitDoesNotHaveEncounterOfType(PihEmrConfigConstants.ENCOUNTERTYPE_ANC_FOLLOWUP_UUID),
+                            patientDoesNotHaveEncounterOfTypeDuringProgramEnrollment(PihEmrConfigConstants.ENCOUNTERTYPE_ANC_INTAKE_UUID, PihEmrConfigConstants.PROGRAM_PREGNANCY_UUID),
```

Comment:
The supporting code looks good, but not sure if this expression will work for all cases?  There is no notion of time here - the implementation below appears to check the patient's active program enrollments.  But what if this is opened in a retrospective context?

Presumably what we are trying to figure out is relative to a particular date - either the current date if no visit is present in the model, or the visit date if it is present in the model.  And as of that date:
* Is the patient enrolled in a particular program on that date
* Did they have an encounter of a particular type at any point between the dateEnrolled and dateCompleted of that enrollment (whether that enrollment is currently active or not).

Reply:
OK, then we would need to expand the appcontextmodel to include all programs the patient has ever been enrolled. I was working under the impression that this requirement of limiting the number of encounter types applies to only active program enrollments. 
Why did you limit here the list of programs included in the model to only activePrograms?
https://github.com/openmrs/openmrs-module-coreapps/blob/master/omod/src/main/java/org/openmrs/module/coreapps/contextmodel/AppContextModelGenerator.java#L106